Sources/Modules/FilesModule.swift

```diff
@@ -242,26 +242,35 @@ public class FilesModule {
     ///     can be uploaded and whether specified name is unique and won't cause conflicts.
     ///   - completion: Returns a standard file object or an error if the parentId is invalid or if a file
     ///     name collision occurs.
+    /// - Returns: BoxUploadTask
+    @discardableResult
     public func upload(
         data: Data,
         name: String,
         parentId: String,
         progress: @escaping (Progress) -> Void = { _ in },
         performPreflightCheck: Bool = false,
         completion: @escaping Callback<File>
-    ) {
-        uploadWithPreflightCheck(
-            performCheck: performPreflightCheck,
-            name: name,
-            size: Int64(data.count),
-            parentId: parentId,
-            request: { [weak self] in
-                guard let self = self else {
-                    return
-                }
-                self.upload(data: data, name: name, parentId: parentId, progress: progress, completion: completion)
-            }, completion: completion
+    ) -> BoxUploadTask {
+        let task = BoxUploadTask()
+        task.receiveTask(
+            uploadWithPreflightCheck(
+                performCheck: performPreflightCheck,
+                name: name,
+                size: Int64(data.count),
+                parentId: parentId,
+                request: { [weak self] in
+                    guard let self = self else {
+                        return
+                    }
```

Comment:
Is it expected that uploads don't make progress unless you retain the Files module?